distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/cometd/CometDClient.java

```diff
@@ -123,51 +159,93 @@ public void subscribe(String channel) {
+     */
     public List<String> getMessages(String channel) {
         verifyConnected();
         verifySubscribed();
         return messageListeners.stream()
-                .filter(l -> l.getChannel().equals(channel))
-                .flatMap(l -> l.getMessages().stream())
+                .filter(l -> l.getChannel()
+                        .equals(channel))
+                .flatMap(l -> l.getMessages()
+                        .stream())
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Gets the list of messages received on a given channel in time ascending order, i.e., from
+     * oldest to most recent.
+     *
+     * @param channel channel name
+     * @return list of message received since the client was started
+     */
     public List<String> getMessagesInAscOrder(String channel) {
         List<String> messages = getMessages(channel);
         Collections.sort(messages, new AscendingTimestampComparator());
         return messages;
     }
 
+    /**
+     * Gets the CometD client ID.
+     *
+     * @return CometD client ID
+     */
     public String getClientId() {
         verifyConnected();
         return bayeuxClient.getId();
     }
 
+    /**
+     * Gets the list of messages received on all channels.
+     *
+     * @return list of message received since the client was started
+     */
     public List<String> getAllMessages() {
         verifyConnected();
         verifySubscribed();
         return messageListeners.stream()
-                .flatMap(l -> l.getMessages().stream())
+                .flatMap(l -> l.getMessages()
+                        .stream())
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Gets the list of messages received on all channels in time ascending order, i.e., from
+     * oldest to most recent.
+     *
+     * @return list of message received since the client was started
+     */
     public List<String> getAllMessagesInAscOrder() {
         List<String> messages = getAllMessages();
         Collections.sort(messages, new AscendingTimestampComparator());
         return messages;
     }
 
+    /**
+     * Un-subscribes from a channel. Un-subscribing from the same channel multiple has no effect.
+     *
+     * @param channel channel name
+     */
     public void unsubscribe(String channel) {
         verifyConnected();
-        org.cometd.bayeux.client.ClientSessionChannel.MessageListener messageListener = messageListeners
-                .stream()
-                .filter(l -> l.getChannel().equals(channel))
-                .findFirst()
-                .get();
-        bayeuxClient.getChannel(channel).unsubscribe(messageListener);
+        org.cometd.bayeux.client.ClientSessionChannel.MessageListener messageListener =
+                messageListeners.stream()
+                        .filter(l -> l.getChannel()
+                                .equals(channel))
+                        .findFirst()
+                        .get();
```

Comment:
Should protect from a non-existent channel (bad named passed in) which would result in a `NoSuchElementException`. The following is cleaner and safer:
```java
messageListeners.stream()
        .filter(l -> l.getChannel()
                .equals(channel))
        .findFirst()
        .ifPresent(ml -> {
            bayeuxClient.getChannel(channel)
                    .unsubscribe(ml);
            messageListeners.remove(ml);
        });
```